luaui/Widgets/gui_comselect.lua

```diff
@@ -0,0 +1,128 @@
+function widget:GetInfo()
+	return {
+		name = "Commander Selector",
+		desc = "Allows for proper keybind selection of any flavour of commanders",
+		author = "Hornet, with some ZK raiding",
+		date = "03 June 2024",
+		license = "GNU GPL, v2 or later",
+		layer = 1001,
+		enabled = true,
+	}
+end
+
+
+
+local myTeamID = Spring.GetMyTeamID()
+local spGetUnitTeam = Spring.GetUnitTeam
+local spGetMyTeamID = Spring.GetMyTeamID
+local spSendCommands = Spring.SendCommands
+
+local specOld = false
+local spec = false
+local team
+
+
+local commanderList = {}
+local commanderIds = {}
+for udid, ud in pairs(UnitDefs) do
+	if ud.customParams and ud.customParams.iscommander then
+		commanderIds[udid] = udid
+	end
+end
+
+function widget:MetaUnitAdded(unitID, unitDefID, unitTeam)
+	if (not myTeamID or unitTeam ~= myTeamID) then
+		return
+	end
+	if commanderIds[unitDefID] then
+		table.insert(commanderList, unitID)
+		--Spring.Echo('hornet comsel adding com')
+	end
+end
+
+
+function widget:MetaUnitRemoved(unitID, unitDefID, unitTeam)
+	if commanderList[unitID] then
+		commanderList[unitID] = nil
+	end
```

Comment:
This is an array so indexed with 1, 2, 3... not unitIDs.

It would probably be a good idea to make it indexed by unitID for simplicity, but keep in mind iterating over such table has an unspecified order and it's important for the order to stay consistent across multiple selections.
```suggestion
	if not commanderIds[unitDefID] then
		return
	end

	local found = commanderList:removeFirst(unitID)

	-- if not found then
	-- 	error("unit is a commander but was not on the list!", unitID)
	-- end
```